lib/grpc/transport/http2.ex

```diff
@@ -29,6 +29,26 @@ defmodule GRPC.Transport.HTTP2 do
     }
   end
 
+  @spec server_trailers(integer, String.t(), [Google.Protobuf.Any.t()] | nil) :: map
```

Comment:
This positional thing feels weird - both status message and status details are optional, there's no requirement that you can't pass status details if you didn't pass status message. The only requirement is that status details should only be provided if the status is not `OK`.[^1]

I would either:

1. make the `status` field a required positional argument, and the second argument a map with any combination of `message` and/or `details`, OR
2. make this function take a single argument - a map where `status` is a required key and the other two are optional.

[^1]: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md